pkg/util/ensure/ensure.go

```diff
@@ -0,0 +1,159 @@
+package ensure
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	projectv1 "github.com/openshift/api/project/v1"
+	securityv1 "github.com/openshift/api/security/v1"
+	securityclient "github.com/openshift/client-go/security/clientset/versioned"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+type Ensure interface {
+	EnsureNamespace(ns string) error
+	EnsureScc(ctx context.Context) (*securityv1.SecurityContextConstraints, error)
+	ApplyDaemonSet(ds *appsv1.DaemonSet) error
+	ApplyConfigMap(cm *v1.ConfigMap) error
+	ApplySecret(s *v1.Secret) error
+	ApplyServiceAccount(sa *v1.ServiceAccount) error
+	EnsureSccCreate(scc *securityv1.SecurityContextConstraints) error
+}
```

Comment:
I think it is a good start, but this somehow does not read yet right.
1. Ensure.Ensure is over verbose. 
2. What is the underlying diff in Ensure and Apply? Should we just simplify with `Ensure.Object` (using full Kubernetes name)